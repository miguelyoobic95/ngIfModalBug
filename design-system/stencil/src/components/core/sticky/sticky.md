---
name: Sticky
category: Core Components
---

Sticky components are used to for elements which remain at the top of the viewport, like a sticky navigation.

## Attributes
|Name|Type|Default|Description|
|---|---|---|---|
|`top`|string|`null`|Set the offset top of the sticky Element. Set it to 0 if you don't want any offset|
|`bottom`|string|`null`|Set the offset bottom of the sticky Element. Set it to 0 if you don't want any offset|

## Top Attribute

```yoo-sticky-top.html

    <div class="container">
        <p>
            This is a random container in a page 
            <br/>
            to test the sticky element.
            <br/>
            I hop it is working as expected.
            <br/>
            Call me if it doesn't
        </p>
        <p>
            This is a random container in a page 
            <br/>
            to test the sticky element.
            <br/>
            I hop it is working as expected.
            <br/>
            Call me if it doesn't
        </p>
        <yoo-sticky top="0">
            <div >
                <span>Bonjour</span>
            </div>
        </yoo-sticky>
        <p>
            This is a random container in a page 
            <br/>
            to test the sticky element.
            <br/>
            I hop it is working as expected.
            <br/>
            Call me if it doesn't
        </p>
        <p>
            This is a random container in a page 
            <br/>
            to test the sticky element.
            <br/>
            I hop it is working as expected.
            <br/>
            Call me if it doesn't
        </p>
    <div>
```
```yoo-sticky-top.css

.container {
    overflow: auto;
    height: 200px;
}

span {
    background: green;
    color: red;
}

```

## Bottom Attribute

```yoo-sticky-bottom.html

    <div class="container">
        <p>
            This is a random container in a page 
            <br/>
            to test the sticky element.
            <br/>
            I hop it is working as expected.
            <br/>
            Call me if it doesn't
        </p>
        <p>
            This is a random container in a page 
            <br/>
            to test the sticky element.
            <br/>
            I hop it is working as expected.
            <br/>
            Call me if it doesn't
        </p>
        <yoo-sticky bottom="10%">
            <div >
                <span>Bonjour</span>
            </div>
        </yoo-sticky>
        <p>
            This is a random container in a page 
            <br/>
            to test the sticky element.
            <br/>
            I hop it is working as expected.
            <br/>
            Call me if it doesn't
        </p>
        <p>
            This is a random container in a page 
            <br/>
            to test the sticky element.
            <br/>
            I hop it is working as expected.
            <br/>
            Call me if it doesn't
        </p>
    <div>
```
```yoo-sticky-bottom.css

.container {
    overflow: auto;
    height: 200px;
}

span {
    background: green;
    color: red;
}

```

