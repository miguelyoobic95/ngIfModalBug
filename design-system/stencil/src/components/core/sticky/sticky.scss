

:host {
    .outer-container {
        position: sticky;
    }
}