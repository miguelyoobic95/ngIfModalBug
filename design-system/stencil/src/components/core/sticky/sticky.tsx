import { Component, Element, Prop } from '@stencil/core';

@Component({
    tag: 'yoo-sticky',
    styleUrl: 'sticky.scss',
    scoped: true
})
export class YooStickyComponent {

    @Prop() top: string = null;
    @Prop() bottom: string = null;

     @Element() host: HTMLStencilElement;

    getTranslateStyle(): any {
        return {position: 'sticky', top: this.top, bottom: this.bottom};
    }

    render(): JSX.Element {
        return (
            <div class="outer-container" style={this.getTranslateStyle()}>
                <slot/>
            </div>
        );
    }
}
