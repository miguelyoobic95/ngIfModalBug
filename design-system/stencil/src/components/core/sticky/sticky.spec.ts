import { TestWindow } from '@stencil/core/testing';
import { YooStickyComponent } from './sticky';

//ToDo What if passed with 0 tabs?
//ToDo What if selected tab is in dropdown?
//ToDo What if tab names do not match slot names? (use index instead? )

describe('YooTabsComponent', () => {
    it('should build', () => {
        expect(new YooStickyComponent()).toBeTruthy();
    });

    describe('Rendering', () => {
        it('Should render', async () => {
            let window = new TestWindow();
          let element = await window.load({
                components: [YooStickyComponent],
                html: `<yoo-sticky></yoo-sticky>`
            });
            expect(element).toMatchSnapshot();
        });

        it('Should render with a slot', async () => {
            let window = new TestWindow();
          let element = await window.load({
                components: [YooStickyComponent],
                html: ` <yoo-sticky>
                            <div >
                                <span>Sticky Slot</span>
                            </div>
                        </yoo-sticky>`
            });
            expect(element).toMatchSnapshot();
        });

        it('Should render with top and bot prop', async () => {
            let window = new TestWindow();
          let element = await window.load({
                components: [YooStickyComponent],
                html: ` <yoo-sticky top="0px" bottom="40%">
                            <div >
                                <span>Sticky Slot</span>
                            </div>
                        </yoo-sticky>`
            });
            expect(element.top).toEqual('0px');
            expect(element.bottom).toEqual('40%');
        });

    });

});
