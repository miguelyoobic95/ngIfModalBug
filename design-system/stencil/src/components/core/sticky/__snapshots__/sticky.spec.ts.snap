// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YooTabsComponent Rendering Should render 1`] = `
<yoo-sticky
  class="hydrated"
  data-yoo-sticky-host=""
>
  
  <div
    class="outer-container"
    data-yoo-sticky=""
    data-yoo-sticky-slot=""
    style="position: sticky;"
  >
    
  </div>
</yoo-sticky>
`;

exports[`YooTabsComponent Rendering Should render with a slot 1`] = `
<yoo-sticky
  class="hydrated"
  data-yoo-sticky-host=""
>
  
  
  
  
  <div
    class="outer-container"
    data-yoo-sticky=""
    data-yoo-sticky-slot=""
    style="position: sticky;"
  >
    
    
                            
    <div>
      
                                
      <span>
        Sticky Slot
      </span>
      
                            
    </div>
    
                        
  </div>
</yoo-sticky>
`;
